File: 739.DailyTemperatures.java
class Solution {
    public int[] dailyTemperatures(int[] tmp) {
        Stack<int[]> stack = new Stack<>();
        int[] ans = new int[tmp.length];

        for(int i = 0; i < tmp.length; i++){
            while(!stack.isEmpty() && tmp[i] > stack.peek()[0]){
                int[] curr = stack.pop();
                ans[curr[1]] = i - curr[1];
            }
            stack.push(new int[]{tmp[i], i});
        }
        return ans;
    }
}

File: 739.DailyTemperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        res = [0] * len(temperatures)
        stack = [] # temp and index
        
        for idx, t in enumerate(temperatures):
            
            while stack and t > stack[-1][0]:
                topTemp, topIndex = stack.pop()
                res[topIndex] = idx - topIndex

            stack.append((temperatures[idx], idx))
            

        return res
